File: app/src/main/java/com/absinthe/kage/device/cmd/SetPlayingPositionCommand.java
package com.absinthe.kage.device.cmd;

import com.absinthe.kage.connect.protocol.IpMessageConst;
import com.absinthe.kage.device.Command;
import com.absinthe.kage.device.CommandBuilder;
import com.absinthe.kage.device.client.Client;

public class SetPlayingPositionCommand extends Command {

    public static final int LENGTH = 2;

    public int position;

    public SetPlayingPositionCommand() {
        cmd = IpMessageConst.RESPONSE_SET_PLAYBACK_PROGRESS;
    }

    @Override
    public String pack() {
        return new CommandBuilder()
                .with(this)
                .append(String.valueOf(position))
                .build();
    }

    @Override
    public void doWork(Client client, String received) {

    }

    @Override
    public boolean parseReceived(String received) {
        return false;
    }
}

File: app/src/main/java/com/absinthe/kage/device/cmd/SetPlayingPositionCommand.kt
package com.absinthe.kage.device.cmd

import com.absinthe.kage.connect.protocol.IpMessageConst
import com.absinthe.kage.device.Command
import com.absinthe.kage.device.CommandBuilder
import com.absinthe.kage.device.client.Client

class SetPlayingPositionCommand : Command() {
    @JvmField
    var position = 0

    override fun pack(): String {
        return CommandBuilder()
                .with(this)
                .append(position.toString())
                .build()
    }

    override fun doWork(client: Client, received: String) {}

    override fun parseReceived(received: String): Boolean {
        return false
    }

    companion object {
        const val LENGTH = 2
    }

    init {
        cmd = IpMessageConst.RESPONSE_SET_PLAYBACK_PROGRESS
    }
}